lib/record_store/zone.rb

```diff
@@ -272,6 +273,16 @@ def validate_provider_can_handle_zone_records
       end
     end
 
+    def validate_zone_not_delegated_to_another_provider
+      ns_records = records.select { |record| record.is_a?(Record::NS) }
+
+      ns_records.each do |records|
+        unless record.contains("nsone.net.") || record.contains(".dnsimple.com.")
+          errors.add(:records, "Warning, a zone delegation exists for this domain, this record will have no effect")
+        end
+      end
+    end
```

Comment:
So if I understand correctly, this:
- Selects all NS records in the file
- Checks that any of those NS records is delegated to something else than NS1 and DNSimple
- Returns an error if it's the case

There's a few issues there:
1. This will complain as soon as we add any NS record to the zone. I would delegate `a.example.com` and this would complain right away, even if there is no record being shadowed by the delegation
2. This considers that NS1 and DNSimple records are always fine, and thus would ignore `b.a.example.com` even if there is a delegation for `a.example.com`, thus allowing me to add the record in the wrong zone, and to be shadowed

The first element to consider here is that we don't really care about the provider. The way things are structured in the `record_store` allows us to consider that as long as there is a delegation, I can error on any kind of record shadowed by that delegation. If that delegation brings back to the same providers, then I will have a different zone file to store the information. Putting the records in the wrong file would have them being shadowed and not working.

We thus need to:
- Given a list of records `records`
- List all NS records `ns_records` from `records`, except for those that are for the current zone (`example.com`)
- For each record in `records`, check if the `fqdn` of that record ends with any of the `ns_records`'s `fqdn`
- If they do: there's an error as that record is shadowed, as the subzone has been delegated

Example of case that's an issue:

```
example.com:
  config:
    providers:
      - SomeProvider1
      - SomeProvider2
  records:
    - type: NS
       fqdn: example.com.
       ttl: 3600
       nsdname: ns1.something.com.
    - type: NS
       fqdn: example.com.
       ttl: 3600
       nsdname: ns2.something.com.

    - type: CNAME
       fqdn: b.example.com.
       ttl: 3600
       cname: cname.example.com.

    - type: NS
       fqdn: a.example.com.
       ttl: 3600
       nsdname: ns1.somethingelse.com.

    - type: CNAME.   ##### THIS RECORD IS SHADOWED
       fqdn: b.a.example.com.
       ttl: 3600
       cname: cname.example.com.
```